books/views.py

```diff
@@ -0,0 +1,94 @@
+# Create your views here.
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+
+
+from models import Books, Authors, Tags, Comments
+from rest_framework import viewsets
+from serializers import BookSerializer, AuthorSerializer, TagsSerializer, CommentSerializer
+
+
+class BookList(viewsets.ModelViewSet):
+    """
+    API endpoint that allows books to be viewed or edited.
+    """
+    queryset = Books.objects.all()
+    serializer_class = BookSerializer
+
+
+class AuthorViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows authors to be viewed or edited.
+    """
+    queryset = Authors.objects.all()
+    serializer_class = AuthorSerializer
+
+
+class TagViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows tags to be viewed or edited.
+    """
+    queryset = Tags.objects.all()
+    serializer_class = TagsSerializer
+
+
+class CommentViewSet(viewsets.ViewSet):
+    """
+    API endpoint that allows comments to be viewed or edited.
+    """
+    serializer_class = CommentSerializer
+
+    def list(self, request, book_id):
+        comments = Comments.objects.filter(id_book=book_id)
+        serializer = CommentSerializer(comments, many=True, context={'request': request})
+        return Response(serializer.data)
+
+    def create(self, request, book_id, format=None):
+        serializer = CommentSerializer(data=request.data, context={'request': request})
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def retrieve(self, request, book_id, pk=None):
+        comments = Comments.objects.filter(id_book=book_id)
+        comment = get_object_or_404(comments, pk=pk)
+        serializer = CommentSerializer(comment, context={'request': request})
+        return Response(serializer.data)
+
+    def update(self, request, book_id, pk=None):
+        queryset = Comments.objects.all()
+        comment = get_object_or_404(queryset, pk=pk)
```

Comment:
You can use `comment = get_object_or_404(Comments, pk=pk, id_book=book_id)`
I think it's better to check book_id as well.